Datamigratie.Server/Features/MigrateZaak/MigrateZaakService.cs

```diff
@@ -141,6 +191,78 @@ await CreateAndLinkDocumentAsync(
                 token);
         }
 
+        /// <summary>
+        /// Migrates all documents with their versions in the correct order.
+        /// First version is created, next versions update the same document (OpenZaak auto-increments version).
+        /// </summary>
+        private async Task MigrateDocumentsAsync(DetZaak detZaak, OzZaak createdZaak, Uri informatieObjectType, CancellationToken token)
+        {
+            foreach (var document in detZaak?.Documenten ?? [])
+            {
+                var sortedVersions = document.DocumentVersies.OrderBy(v => v.Versienummer).ToList();
+                
+                string? documentUrl = null;
+                
+                for (var i = 0; i < sortedVersions.Count; i++)
+                {
+                    var versie = sortedVersions[i];
+                    var isFirstVersion = i == 0;
+                    
+                    try
+                    {
+                        var ozDocument = MapToOzDocument(document, versie, informatieObjectType);
+                        
+                        if (isFirstVersion)
+                        {
+                            // create new document and link to zaak
+                            var savedDocument = await _openZaakApiClient.CreateDocument(ozDocument);
+                            documentUrl = savedDocument.Url;
```

Comment:
I think it's risky to use the complete returned url. Sometimes different domains are used internally and externally in PodiumD. Probably better to extract the guid and create the put path in the apiclient